src/reconcile/rec_visibility.c

```diff
@@ -933,6 +932,7 @@ __rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_UPDATE *first_up
               *write_prepare && upd->prepare_state == WT_PREPARE_INPROGRESS &&
                 prepare_rollback_tombstone->next == upd);
             upd_select->upd = upd;
+            prepare_rollback_tombstone = NULL;
```

Comment:
Can we create a separate test case for this issue?

Reply:
added `test_prepare40`